presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/QueryBuilder.java

```diff
@@ -278,4 +293,23 @@ private static void bindValue(Object value, Type type, List<TypeAndValue> accumu
         checkArgument(isAcceptedType(type), "Can't handle type: %s", type);
         accumulator.add(new TypeAndValue(type, value));
     }
+
+    private static long toLong(Object value)
+    {
+        if (value instanceof Long) {
```

Comment:
@electrum 
For now, the value for `tinyint`, `smallint`, `integer`, and `bigint` from `TupleDomain` should always be `Long` due to https://github.com/prestodb/presto/blob/c7bf63a2138c2e61f0edb307cefbf6b35fba25c2/presto-spi/src/main/java/com/facebook/presto/spi/predicate/Utils.java#L32 (`tinyint`, `smallint`, `integer`, and `bigint` is backed by `long`). 

I add this just in case if the above restriction is removed in the future, the casting failure exception won't be too surprising.

I could remove this and always cast the value to `Long` for now.